assets/js/modules/analytics-4/components/widgets/TopConvertingTrafficSourceWidget.stories.js

```diff
@@ -0,0 +1,143 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Internal dependencies
+ */
+import {
+	provideKeyMetrics,
+	provideModules,
+} from '../../../../../../tests/js/utils';
+import { withWidgetComponentProps } from '../../../../googlesitekit/widgets/util';
+import WithRegistrySetup from '../../../../../../tests/js/WithRegistrySetup';
+import TopConvertingTrafficSourceWidget from './TopConvertingTrafficSourceWidget';
+import { CORE_USER } from '../../../../googlesitekit/datastore/user/constants';
+import { MODULES_ANALYTICS_4 } from '../../datastore/constants';
+import {
+	getAnalytics4MockResponse,
+	provideAnalytics4MockReport,
+} from '../../utils/data-mock';
+import { replaceValuesInAnalytics4ReportWithZeroData } from '../../../../../../.storybook/utils/zeroReports';
+
+const reportOptions = {
+	compareStartDate: '2020-07-14',
+	compareEndDate: '2020-08-10',
+	startDate: '2020-08-11',
+	endDate: '2020-09-07',
+	dimensions: [ 'sessionDefaultChannelGroup' ],
+	metrics: [
+		{
+			name: 'sessionConversionRate',
+		},
+	],
+	limit: 1,
+	orderBy: 'sessionConversionRate',
+};
+
+const WidgetWithComponentProps = withWidgetComponentProps(
+	'kmAnalyticsTopTrafficSource'
+)( TopConvertingTrafficSourceWidget );
+
+const Template = ( { setupRegistry, ...args } ) => (
+	<WithRegistrySetup func={ setupRegistry }>
+		<WidgetWithComponentProps { ...args } />
+	</WithRegistrySetup>
+);
+
+export const Ready = Template.bind( {} );
+Ready.storyName = 'Ready';
+Ready.args = {
+	setupRegistry: ( registry ) => {
+		provideAnalytics4MockReport( registry, reportOptions );
+	},
+};
+Ready.scenario = {
+	label: 'KeyMetrics/TopConvertingTrafficSourceWidget/Ready',
+};
+
+export const Loading = Template.bind( {} );
+Loading.storyName = 'Loading';
+Loading.args = {
+	setupRegistry: ( { dispatch } ) => {
+		dispatch( MODULES_ANALYTICS_4 ).startResolution( 'getReport', [
+			reportOptions,
+		] );
+	},
+};
+Loading.scenario = {
+	label: 'KeyMetrics/TopConvertingTrafficSourceWidget/Loading',
+};
+Loading.decorators = [
+	( Story ) => {
+		// Ensure the animation is paused for VRT tests to correctly capture the loading state.
+		return (
+			<div className="googlesitekit-vrt-animation-paused">
+				<Story />
+			</div>
+		);
+	},
+];
+
+export const ZeroData = Template.bind( {} );
+ZeroData.storyName = 'Zero Data';
+ZeroData.args = {
+	setupRegistry: ( { dispatch } ) => {
+		dispatch( MODULES_ANALYTICS_4 ).receiveGetReport(
+			replaceValuesInAnalytics4ReportWithZeroData(
+				getAnalytics4MockResponse( reportOptions )
+			),
+			{
+				options: reportOptions,
+			}
+		);
+	},
+};
+ZeroData.scenario = {
+	label: 'KeyMetrics/TopConvertingTrafficSourceWidget/ZeroData',
+};
+
+export default {
+	title: 'Key Metrics/TopConvertingTrafficSourceWidget',
+	decorators: [
+		( Story, { args } ) => {
+			const setupRegistry = ( registry ) => {
+				global._googlesitekitUserData.isUserInputCompleted = false;
```

Comment:
Can we receive this directly instead?

Reply:
Thanks for calling this out. To be honest, this is a piece of boilerplate I'd copied from another Key Metrics widget story, having seen it in a number of them. I should have questioned it - on closer inspection it turns out simply not to be needed, here, or in a number of the other widget stories. I've removed it from all the places where I've identified it's not needed.

Reply:
Thanks – I recently noticed the same on [another PR](https://github.com/google/site-kit-wp/pull/7255#discussion_r1254771503) and @nfmohit helped clarify that this was being done due to a flaw in the resolver where the action to receive this wasn't preventing the resolver from running when we were setting it to `false` so it would always source the value from the global. This has been fixed now so we can use the action with any boolean 👍

Reply:
Thanks for the detail on this one. So it looks like we can generally avoid setting this global in future in tests/stories, but no further changes are needed on this PR as the value doesn't need to be set for the updated stories anyway, if I have the correct understanding of things here?

Reply:
Sounds right to me, thanks 👍